### PR TITLE
spriv-val: Vulkan image gather constant component

### PR DESCRIPTION
**Pull request by sfricke-samsung**
VUID-StandaloneSpirv-OpImageGather-04664

### CODE REVIEW

File: source/val/validate_image.cpp [hunk 1]
```diff
@@ -1457,12 +1457,22 @@ spv_result_t ValidateImageGather(ValidationState_t& _,
...
-    const uint32_t component_index_type = _.GetOperandTypeId(inst, 4);
+    const uint32_t component = inst->GetOperandAs<uint32_t>(4);
+    const uint32_t component_index_type = _.GetTypeId(component);
     if (!_.IsIntScalarType(component_index_type) ||
         _.GetBitWidth(component_index_type) != 32) {
       return _.diag(SPV_ERROR_INVALID_DATA, inst)
              << "Expected Component to be 32-bit int scalar";
     }
+    if (spvIsVulkanEnv(_.context()->target_env)) {
+      if (!spvOpcodeIsConstant(_.GetIdOpcode(component)) &&
```
Comment (alan-baker) [inline source/val/validate_image.cpp:1468]: This covers spec constants too.

**Review by alan-baker — CHANGES_REQUESTED** [review]

File: source/val/validate_image.cpp (hunk 1 above)
Comment (sfricke-samsung) [reply source/val/validate_image.cpp:1468]: The Vulkan spec says 

> must be the <id> of a constant instruction

So does that not suppose to include spec constants as well?

Comment (alan-baker) [reply source/val/validate_image.cpp:1468]: It does. `spvOpcodeIsConstant` checks for spec constants too I meant. You can drop the specific spec constant check.

Comment (sfricke-samsung) [reply source/val/validate_image.cpp:1468]: ... 🤦 I didn't see the `spvOpcodeIsSpecConstant` I still left, yes will definitely remove, thanks for catching

**Review by alan-baker — APPROVED** [review]

**Pull request merged**